src/main/java/org/apache/commons/compress/archivers/zip/ZipFile.java

```diff
@@ -820,7 +839,13 @@ public Enumeration<ZipArchiveEntry> getEntries() {
      * @since 1.6
      */
     public Iterable<ZipArchiveEntry> getEntries(final String name) {
-        return nameMap.getOrDefault(name, ZipArchiveEntry.EMPTY_LINKED_LIST);
```

Comment:
You mentioned a potential 30% slowdown in this method in return for additional safety, which may or may not be a fair tradeoff. But I'm a bit concerned that this change potentially breaks the current behaviour. As you say, this currently returns a modifiable LinkedList, which some people may (though I hope not) depend on? Because it requires an explicit cast, I'm not too worried.

Reply:
I think the slowdown is only in case there are duplicit entries which is extremely unlikely.  I'd still simplify the code, as according to earlier comment.